### ISSUE TITLE
ReduceLROnPlateau does not support `maximize-best-checkpoint-metric`

### ISSUE THREAD

**Issue by pietruh** [pytorch/fairseq#2205]
## 🐛 Bug

In the Torch optimizer package, [ReduceLROnPlateau](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html#ReduceLROnPlateau) is implemented with default `mode=min`, but in [Fairseq wrapper](https://github.com/pytorch/fairseq/blob/1e40a48037eefc5ceb7672ea0fa11db629a35113/fairseq/optim/lr_scheduler/reduce_lr_on_plateau.py#L36), this probably should be called with `mode=max` in the situation when training is performed with the `maximize-best-checkpoint-metric` switch. In the current setup, it is reducing `lr` as the training progresses. The mode can be easily inferred based on available args in the fairseq's linked code line.

### Expected behavior

I suppose that the expected behaviour is to change the mode of torch optim to `max` when using `maximize-best-checkpoint-metric` parameter, but I may be missing something fundamental here.

### Environment

Recent codebases of Pytorch and Fairseq
